src/ast/mod.rs

```diff
@@ -102,10 +102,26 @@ fn write_tree_view_recursive<'arena, Node>(
     }
 }
 
+/// The specification of an NodeClass that Class can have
+pub trait NodeClass<'arena>: Copy + std::fmt::Debug + Eq + std::hash::Hash {
+    /// Gets the [`char`] that would have been used to create this value
+    fn to_char(self) -> char;
+
+    /// Returns the name of this value
+    fn name(self) -> &'static str;
+
+    /// Returns Option<Self> from ['char']
+    /// If [`char`] is valid will return [`Some(char)`]
+    /// If [`char`] is invalid will return [`None`]
+    fn from_char(c: char) -> Option<Self>;
```

Comment:
When Rust creates documentation, it will mash all newlines together.  So this would end up as `Returns Option<Self> from ['char`] If [`char`] is valid will return [`Some(char)`] If [`char`] is invalid will return [`None`]`.

This could probably be a 1-liner, like 
```rust
/// Creates a `NodeClass` from a [`char`], returning [`None`] if invalid.
```